tests/primer/primer_tool.py

```diff
@@ -112,11 +130,79 @@ def _handle_run_command(self) -> None:
+
+        final_main_dict: PackageMessages = {}
+        for package, messages in main_dict.items():
+            final_main_dict[package] = []
+            for message in messages:
+                try:
+                    new_dict[package].remove(message)
+                except ValueError:
+                    final_main_dict[package].append(message)
+
+        self._create_comment(final_main_dict, new_dict)
+
+    def _create_comment(
+        self, all_missing_messages: PackageMessages, all_new_messages: PackageMessages
+    ) -> None:
+        comment = ""
+        for package, missing_messages in all_missing_messages.items():
+            new_messages = all_new_messages[package]
+            package_data = self.packages[package]
+
+            if not missing_messages and not new_messages:
+                continue
+
+            comment += f"\n\n**Effect on [{package}]({self.packages[package].url}):**\n"
+
+            if missing_messages:
+                comment += "The following messages are no longer emitted:\n"
+            count = 1
+            for message in missing_messages:
+                comment += (
+                    f"{count}) {message['symbol']} on line {message['line']} here:\n"
+                )
+                filepath = str(message["path"]).replace(
+                    str(package_data.clone_directory), ""
+                )
+                comment += f"{package_data.url}/blob/{package_data.branch}/{filepath}#L{message['line']}\n"
+                count += 1
+            comment += "\n"
+
+            count = 1
+            if new_messages:
+                comment += "The following messages are now emitted:\n"
+            for message in new_messages:
+                comment += (
+                    f"{count}) {message['symbol']} on line {message['line']} here:\n"
+                )
+                filepath = str(message["path"]).replace(
+                    str(package_data.clone_directory), ""
+                )
+                comment += f"{package_data.url}/blob/{package_data.branch}/{filepath}#L{message['line']}\n"
+                count += 1
+
+        if comment == "":
+            comment = "🤖 According to the primer, this change has **no effect** on the checked open source code. 🤖🎉"
+        else:
+            comment = (
+                "🤖 **Effect of this PR on checked open source code:** 🤖\n\n" + comment
+            )
+
+        with open("comment.txt", "w", encoding="utf-8") as f:
+            f.write(comment)
+
     def _lint_package(self, data: PackageToLint) -> list[dict[str, str | int]]:
         # We want to test all the code we can
         enables = ["--enable-all-extensions", "--enable=all"]
         # Duplicate code takes too long and is relatively safe
-        disables = ["--disable=duplicate-code"]
+        # TODO: Find a way to allow cyclic-import and compare output correctly
+        disables = ["--disable=duplicate-code,cyclic-import"]
```

Comment:
This is a sign we shouldn't be using multiprocessing. As I think you might be aware, parallel mode is currently nondeterministic/unsafe in pylint, and it's not just `cyclic-import`, it's potentially anywhere given #4874 and other issues we haven't fully investigated. I'd like to look at that for 2.15, as it's major technical debt, but until then, I think we need to use `--jobs=1` for the primer runs. Sorry I didn't catch this on the other PR. Then we can revert this change here.

If we want to speed up primer runs we could use parallel CI jobs instead trying to get pylint to do the heavy lifting.

Reply:
Ah -- I might be wrong! Now that I look at the `Run` call, I suppose we're not using parallel mode. Do we have a GitHub issue that captures this problem with `cyclic-import` without using `multiprocessing`?

Reply:
Possibly https://github.com/PyCQA/pylint/issues/4171 ?

Reply:
Ah, but that's `--jobs=0`, not 1, which is partly what confused me, since that _does_ use parallel mode!

Reply:
Maybe Daniel can speak to the failures he saw and if he was testing locally in parallel mode.